### PR TITLE
json、logfmt parsers support keep raw data

### PR DESCRIPTION
**Pull request by mockerzzz**
* Parser [json、logfmt]支持保留原始数据

### CODE REVIEW

File: parser/logfmt/logfmt_test.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func TestParse(t *testing.T) {
 			expectData: []Data{
 				{
 					"ts":     "2018-01-02T03:04:05.123Z",
-					"lvl":    int64(5),
+					"lvl":    float64(5),
```
Comment (redHJ) [inline parser/logfmt/logfmt_test.go]: 这个为什么要改？

**Comment by redHJ** [comment]
实测一下，打到pandora

**Comment by zoo4778362** [comment]
lgtm

File: parser/logfmt/logfmt.go [hunk 2]
```diff
@@ -115,21 +131,25 @@ func (p *Parser) parse(line string) ([]Data, error) {
...
 	for {
 		ok := decoder.ScanRecord()
 		if !ok {
 			err := decoder.Err()
 			if err != nil {
 				return nil, err
 			}
+			//此错误仅用于当原始数据解析成功但无解析数据时，保留原始数据之用
+			if len(fields) == 0 {
+				return nil, errors.New("field's length is zero,but original data should be reserved.")
```
Comment (redHJ) [inline parser/logfmt/logfmt.go]: no value was parsed after logfmt, will keep origin data in pandora_stash if disable_record_errdata field is false

**Pull request merged**